builders/integrations/oracles/dia.md

```diff
@@ -0,0 +1,289 @@
+
+contract DIARandomOracle {
+  struct Random {
+    string randomness;
+    string signature;
+    string previousSignature;
+  }
+
+  mapping(uint256 => Random) public values;
+  uint256 public lastRound = 0;
+  address public oracleUpdater;
+  event OracleUpdate(string key, uint128 value, uint128 timestamp);
+  event UpdaterAddressChange(address newUpdater);
+
+  constructor() {
+      oracleUpdater = msg.sender;
+  }
+
+  function setRandomValue(
+    uint256 _round,
+    string memory _randomness,
+    string memory _signature,
+    string memory _previousSignature
+  ) public {
+    require(msg.sender == oracleUpdater, "not a updater");
+    require(lastRound < _round, "old round");
+    lastRound = _round;
+    values[_round] = Random(_randomness, _signature, _previousSignature);
+  }
+
+  function getValue(uint256 _round) external view returns (Random memory) {
+    return values[_round];
+  }
+
+  function updateOracleUpdaterAddress(address newOracleUpdaterAddress)
+    public
+  {
+    require(msg.sender == oracleUpdater, "not a updater");
+    oracleUpdater = newOracleUpdaterAddress;
+    emit UpdaterAddressChange(newOracleUpdaterAddress);
+  }
+
+  function getRandomValueFromRound(uint256 _round)
+    external
+    view
+    returns (string memory)
+  {
+    return values[_round].randomness;
+  }
+
+  function getRandomValueFromRoundWithSignature(uint256 _round)
+    external
+    view
+    returns (Random memory)
+  {
+    return values[_round];
+  }
+
+    function getLastRound() public view returns (uint256) {
+    return lastRound;
+  }
+}
+```
+
+!!! note
+    DIA demo oracles are not intended for use in production environments. Developers can request a dedicated, production-ready randomness oracle. To start the request process, you can check out the [Request a Random Oracle](https://docs.diadata.org/products/randomness-oracle/request-a-random-oracle){target=_blank} documentation.
+
+### How to Use the DIA Randomness Oracle {: #how-to-use-the-dia-randomness-oracle }
+
+The steps for accessing a published random value are as follows:
+
+1. Access your randomness oracle smart contract on Moonbeam
+2. Call `getLastRound()`to obtain the ID of the latest published round. You can use the **Read Contract** functionality under the **Contract** tab of the contract on Moonscan to execute this call
+3. Call `getRandomValueFromRound(uint256 _round)` using the obtained round ID. Again, you can use Moonscan to quickly execute this call
+
+The response contains the randomness value.
+
+The signature can also be requested by calling `getRandomValueFromRoundWithSignature(uint256 _round)`, which returns a tuple containing the randomness value, the signature, and the previous signature.
+
+To learn how to deploy a randomness-consuming contract on Moonbeam, please refer to the [Deploying a #Randomness Consuming Smart Contract on #EVM chains with DIA xRandom #Oracle](https://youtu.be/BzN-tBgW-xs){target=_blank} video tutorial.
```

Comment:
```suggestion
To learn how to deploy a randomness-consuming contract on Moonbeam, please refer to the [Deploying a Randomness Consuming Smart Contract on EVM chains with DIA xRandom Oracle](https://youtu.be/BzN-tBgW-xs){target=_blank} video tutorial.
```

Reply:
Not a huge fan of the hashtags in the title

Reply:
Totally fair. I just copied it from the actual title on YouTube, but it's not a big deal, so we can remove them!